### PR TITLE
Upgrade to latest Jest and Babel

### PR DESCRIPTION
**Pull request by janizde**
**This PR is based on https://github.com/babel/generator-babel-plugin/pull/18 and contains commits of both until the merge base changes. Changes of only this PR here: https://github.com/janizde/generator-babel-plugin/compare/bump-node-version...janizde:modernize-stack**

* Upgrades to latest Jest
  * Change `test` scripts
  * Move from `test` to `__tests__`: This might be controversial since with this it changes the structure of the generated package, e.g. projects created with a prior version of the generator cannot generate with a `yo babel-plugin:fixture <fixtureName>` command. Alternatively would have to override jest config in `package.json` or `jest.config.js`
* Move from `var` to `let` and `const` where appropriate
* Upgrade to latest babel
* Add tests for empty inputs
  * Add `getAuthor()` to prevent it from generating `"author": " <>"`

### CODE REVIEW

**Comment by janizde** [comment]
@JLHwung could you have a look at this, it's being around for quite a while. We did the internal tools upgrade here: https://github.com/babel/generator-babel-plugin/pull/18 and it would be great to release both as a major update.

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "url": "https://github.com/thejameskyle"
   },
   "scripts": {
-    "test": "mocha"
+    "test": "jest __tests__/*.js"
```
Comment (JLHwung) [inline package.json:14]: nit
```suggestion
    "test": "jest"
```

**Review by JLHwung — APPROVED** [review]
Look good to me other than nit. Thanks!

File: package.json (hunk 1 above)
Comment (janizde) [reply package.json:14]: Good catch, let's stick with the defaults

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "url": "https://github.com/thejameskyle"
   },
   "scripts": {
-    "test": "mocha"
+    "test": "jest"
```
Comment (JLHwung) [inline package.json:14]: The CI error is related, we can add those plugin templates to `testPathIgnorePatterns `: https://jestjs.io/docs/en/configuration#testpathignorepatterns-arraystring

**Comment by janizde** [comment]
@JLHwung The tests are passing now 👍

**Comment by JLHwung** [comment]
Thanks!

**Comment by janizde** [comment]
@JLHwung Thanks for the quick review and merge! Will this get published as a new major or what is the process there?
